runelite-mixins/src/main/java/net/runelite/mixins/GameScalingMixin.java

```diff
@@ -112,20 +166,26 @@ public Dimension getRealDimensions()
 	@Override
 	public Dimension getStretchedDimensions()
 	{
-		Canvas canvas = getCanvas();
+		gameScaling$checkUpToDate();
 
-		int width = canvas.getParent().getWidth();
-		int height = canvas.getParent().getHeight();
-
-		if (cachedStretchedDimensions == null || width != lastCanvasDimensions.width || height != lastCanvasDimensions.height)
+		if (cachedStretchedDimensions == null)
```

Comment:
Same here

Reply:
I don't think that would work in this case

Reply:
if (cachedStetchedDimensions != null)
{
  return cachedStretchedDimensions;
}

Surely it will. Its just kind of hard to read these nested ifs, when basically the entire method is one big if block. But, its just cosmetical improvement so it does not really matter

Reply:
And then do 
```
cachedStretchedDimensions = new Dimension(width, height);
return cachedStretchedDimensions;
``` at the end?

Reply:
yeah

Reply:
I think that looks more confusing

Reply:
Alright